pandas/io/tests/test_sql.py

```diff
@@ -916,19 +918,14 @@ def test_not_reflect_all_tables(self):
             # Verify some things
             self.assertEqual(len(w), 0, "Warning triggered for other table")
 
-    def test_warning_case_insensitive_table_name(self):
+    @mock.patch('pandas.io.sql.SQLDatabase.trigger_case_sensitive_warning')
+    def test_warning_case_insensitive_table_name(self, patch_obj):
```

Comment:
@jreback I haven't seen much monkey patching in the code base but how do you feel about something like this?